pkg/segment/sortindex/sortindex.go

```diff
@@ -658,37 +657,85 @@ func readMetadata(file *os.File) (*metadata, error) {
 	return meta, nil
 }
 
-func SetSortColumns(columnNames []string) error {
-	sortConfig.mu.Lock()
-	defer sortConfig.mu.Unlock()
+func SetSortColumns(indexName string, columnNames []string) error {
+	sortConfigMutex.Lock()
+	defer sortConfigMutex.Unlock()
+
+	dir := filepath.Dir(SortColumnsConfigPath)
+	if err := os.MkdirAll(dir, 0755); err != nil {
+		return fmt.Errorf("failed to create config directory: %v", err)
```

Comment:
nit:
```suggestion
		return fmt.Errorf("SetSortColumns: failed to create config directory: %v", err)
```

There's a few other errors/logs you have that should also include the function name